packages/jsii-pacmak/lib/util.ts

```diff
@@ -6,19 +6,72 @@ import * as path from 'path';
 import * as logging from './logging';
 
 /**
- * Given an npm package directory and a dependency name, returns the package directory of the dep.
- * @param packageDir     the root of the package declaring the dependency.
- * @param dependencyName the name of the dependency to be resolved.
- * @return the resolved directory path.
+ * Find the directory that contains a given dependency, identified by its 'package.json', from a starting search directory
  */
-export function resolveDependencyDirectory(
-  packageDir: string,
+export async function findDependencyDirectory(
   dependencyName: string,
-): string {
-  const lookupPaths = [path.join(packageDir, 'node_modules')];
-  return path.dirname(
-    require.resolve(`${dependencyName}/package.json`, { paths: lookupPaths }),
+  searchStart: string,
+) {
+  // Explicitly do not use 'require("dep/package.json")' because that will fail if the
+  // package does not export that particular file.
+  const entryPoint = require.resolve(dependencyName, {
+    paths: [searchStart],
+  });
```

Comment:
We *could* try `require("dep/package.json")` in first intention (although this could return a different file than what you expect due to import/export maps.

Reply:
Yeah, doesn't seem to be much of a point.